src/js/utils.js

```diff
@@ -156,12 +156,24 @@ function estimateMaxEntropy(str) {
 
   // If all the characters come from one of these common character groups,
   // assume that the group is the domain of possible characters.
-  for (let chr_class in [BIN, DEC, HEX, ALPHA, ALPHANUM, B64, URL]) {
+  for (let chr_class of [BIN, DEC, HEX, ALPHA, ALPHANUM, B64, URL]) {
     let group = chr_class + SEPS;
     // Ignore separator characters when computing entropy. For example, Google
     // Analytics IDs look like "14103492.1964907".
-    if (str.split().every(val => group.includes(val))) {
+
+    // flag to check if each character of input string belongs to the group in question
+    let each_char_in_group = true;
+
+    [...str].forEach(char => {
+      if (!group.includes(char)) {
+        each_char_in_group = false;
```

Comment:
Don't need to keep checking the rest of the group once we find a character that doesn't belong. (`forEach()` doesn't support short-circuiting but we can use a `for...of` instead. Or, we could use a method that short-circuits for us (`every()`), like we originally tried to do.)